### PR TITLE
impr: Replaced volume settings with slider for issue (ryantang35)

### PR DESCRIPTION
**Pull request by ryantang35**
### Description

I replaced the quiet, medium, loud volume settings with a slider. This addresses issue [#4764]. 

### Checks

- [ ] Adding a language or a theme?
    - [ ] If is a language, did you edit `_list.json`, `_groups.json` and add `languages.json`?
    - [ ] If is a theme, did you add the theme.css?
        - Also please add a screenshot of the theme, it would be extra awesome if you do so!
- [ ] Check if any open issues are related to this PR; if so, be sure to tag them below.
- [ ] Make sure the PR title follows the Conventional Commits standard. (https://www.conventionalcommits.org for more info)
- [ ] Make sure to include your GitHub username inside round brackets at the end of the PR title

### CODE REVIEW

File: frontend/src/ts/config.ts [hunk 1]
```diff
@@ -192,7 +192,7 @@ export function setSoundVolume(
   val: MonkeyTypes.SoundVolume,
   nosave?: boolean
 ): boolean {
-  if (!isConfigValueValid("sound volume", val, [["0.1", "0.5", "1.0"]])) {
+  if (!isConfigValueValid("sound volume", val, ["number"])) {
```
Comment (fehmer) [inline frontend/src/ts/config.ts:195]: I think you need to convert users existing config from string to number. There is already a method called  `replaceLegacyValues` for this.

File: frontend/src/ts/types/types.d.ts [hunk 2]
```diff
@@ -1,10 +1,6 @@
 type typesSeparatedWithHash<T> = T | `${T}#${typesSeparatedWithHash<T>}`;
 
 declare namespace MonkeyTypes {
-  type Configuration = import("@backend/types/shared").Configuration;
```
Comment (fehmer) [inline frontend/src/ts/types/types.d.ts:4]: unrelated change? please revert

File: frontend/src/ts/elements/custom-background-filter.ts [hunk 3]
```diff
@@ -19,6 +19,10 @@ const filters = {
     value: 1,
     default: 1,
   },
+  volume: {
```
Comment (fehmer) [inline frontend/src/ts/elements/custom-background-filter.ts:22]: @Miodec is there a better place to put this?

Comment (Miodec) [reply frontend/src/ts/elements/custom-background-filter.ts:22]: Yes, definitely not inside `custom-background-filter`

**Comment by monkeytypegeorge** [comment]
Continuous integration check(s) failed. Please review the failing check's logs and make the necessary changes. https://github.com/monkeytypegame/monkeytype/actions/runs/7093636225
